python/pyarrow/ipc.pxi

```diff
@@ -1192,6 +1198,13 @@ cdef class _RecordBatchFileReader(_Weakrefable):
             raise ValueError("Operation on closed reader")
         return _wrap_read_stats(self.reader.get().stats())
 
+    @property
+    def metadata(self):
+        """
+        File-level custom KeyValueMetadata written via ``ipc.new_file(..., metadata=)``.
+        """
+        return pyarrow_wrap_metadata(self.reader.get().metadata())
```

Comment:
In other metadata properties, i.e: https://github.com/apache/arrow/blob/88815a1f6ffcfbcecdfb7c0863795123ebf1a733/python/pyarrow/types.pxi#L2591-L2595

We return either a dict or None, I think we should do the same to follow consistency on our APIs.
```suggestion
        wrapped = pyarrow_wrap_metadata(self.reader.get().metadata())
        if wrapped is not None:
            return wrapped.to_dict()
        else:
            return wrapped
```

Reply:
Thanks for the feedback @raulcd! I also adjusted the docstring and the test to reflect the change (and added a unit test to check that `pyarrow.ipc.open_file(...).metadata is None` in case nothing was set.)